// @prettier/plugin-pug/dist/options/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CATEGORY_PUG = 'Pug';
function resolveAttributeSeparatorOption(attributeSeparator) {
    switch (attributeSeparator) {
        case 'always':
            return true;
        case 'as-needed':
            return false;
    }
    throw new Error(`Invalid option for pug attributeSeparator. Found '${attributeSeparator}'. Possible options: 'always' or 'as-needed'`);
}
exports.resolveAttributeSeparatorOption = resolveAttributeSeparatorOption;
function formatCommentPreserveSpaces(input, commentPreserveSpaces, pipeless = false) {
    switch (commentPreserveSpaces) {
        case 'keep-leading': {
            let result = '';
            let firstNonSpace = 0;
            for (firstNonSpace; firstNonSpace < input.length && input[firstNonSpace] === ' '; firstNonSpace++) {
                result += ' ';
            }
            result += input.slice(firstNonSpace).trim().replace(/\s\s+/g, ' ');
            return result;
        }
        case 'trim-all': {
            let result = input.trim();
            result = result.replace(/\s\s+/g, ' ');
            if (!pipeless && input[0] === ' ') {
                result = ` ${result}`;
            }
            return result;
        }
        case 'keep-all':
        default:
            return input;
    }
}
exports.formatCommentPreserveSpaces = formatCommentPreserveSpaces;
exports.options = {
    attributeSeparator: {
        since: '1.0.0',
        category: exports.CATEGORY_PUG,
        type: 'choice',
        default: 'always',
        description: 'Change when attributes are separated by commas in tags.',
        choices: [
            {
                value: 'always',
                description: 'Always separate attributes with commas. Example: `button(type="submit", (click)="play()", disabled)`'
            },
            {
                value: 'as-needed',
                description: 'Only add commas between attributes where required. Example: `button(type="submit", (click)="play()" disabled)`'
            }
        ]
    },
    commentPreserveSpaces: {
        since: '1.1.0',
        category: exports.CATEGORY_PUG,
        type: 'choice',
        default: 'keep-all',
        description: 'Change behavior of spaces within comments.',
        choices: [
            {
                value: 'keep-all',
                description: 'Keep all spaces within comments. Example: `//    this  is   a   comment`'
            },
            {
                value: 'keep-leading',
                description: 'Keep leading spaces within comments. Example: `//    this is a comment`'
            },
            {
                value: 'trim-all',
                description: 'Trim all spaces within comments. Example: `// this is a comment`'
            }
        ]
    }
};
